lib/ethereum/client.rb

```diff
@@ -86,13 +86,13 @@ def transfer_to_and_wait(address, amount)
     end
 
 
-    def transfer(key, address, amount)
+    def transfer(key, address, amount, data = "")
```

Comment:
The transfer is not a normal transaction - it is just a value transfer. For contracts, you should use `eth_sendTransaction`.

Reply:
`eth_sendTransaction` is not supported in FUJI testnetwork, also seems Infura.

Reply:
Most third-party node does not keep your private key, so they suggest using `eth_sendRawTransaction` instead of `eth_sendTransaction`

refer to: 

https://stackoverflow.com/questions/68236469/error-returned-error-the-method-eth-sendtransaction-does-not-exist

https://infura.io/docs/ethereum/json-rpc/eth_sendRawTransaction

Reply:
yes, just look at the `transfer` towards the end. it just uses a `Eth::Tx.new()` and sends a raw transaction.

Reply:
thanks for the reply, so can I submit this PR for `transfer` method?  ( the PR is updated with 3 commits now)

Reply:
No, you should not use `transfer` with data.